### ISSUE TITLE
Implement a promise

### ISSUE THREAD

**Issue by advename** [iconfu/svg-inject#18]
Since I'm already suggesting stuff, I would also implement a promise.

I.e. 
DOMContentLoaded > SVGInject > init or function (but only on a resolved callback).

Because currently the issue with both inits is that the svg inject takes around 10ms to inject.

**Comment by advename** [comment]
Yes, exactly, thanks a lot!

**Comment by dasboe** [comment]
implemented in 2ed841d7ae1d1cce3ef4e0701fd054365cd46779

will be part of upcoming release version 1.1.2

**Comment by advename** [comment]
Amazing, could you then maybe add an example to the docs how it would be used for to run a script after all completed injects?

**Comment by dasboe** [comment]
Yes, we will add an example how this can be achieved to the docs. 

In Version 1.2.0 there will be two possible ways how to run scripts after all SVGs have been injected:

(1) Use the options hook `onAllFinish`

```html
<img src="image1.svg" class="injectable" />
<img src="image2.svg" class="injectable" />
<img src="image3.svg" class="injectable" />
```

```js
SVGInject(document.querySelectorAll('img.injectable'), {
  onAllFinish: function() {
    // all three images have been injected
  }
});
```

(2) use the Promise object returned by `SVGInject()`

```html
<img src="image1.svg" class="injectable" />
<img src="image2.svg" class="injectable" />
<img src="image3.svg" class="injectable" />
```

```js
// SVGInject returns a Promise object which can be used to trigger if all injections are completed
SVGInject(document.querySelectorAll('img.injectable')).then(function() {
  // all three images have been injected
});
```

This method only works for [browsers supporting the Promise object](https://caniuse.com/#feat=promises) or any browser with a [Promise polyfill](https://www.npmjs.com/package/promise-polyfill)

Both methods do not provide any parameters for the passed in function. This may be something which we may change later. It would be possible to provide an array of successfully injected SVGs as first parameter and an array of failed to inject `img` elements as second parameter but I'm not so sure if this is really needed.

@advename - could you maybe confirm if this all works for you. It's all up and running in the development branch.

**Comment by iconfu** [comment]
just released Version 1.2.0 which implements the use of a Promise object
